tests/test_client.py

```diff
@@ -0,0 +1,197 @@
+        self.assertEqual(contents, content)
+
+        with pytest.raises(Exception):
+            # assert that the call is to minio2
+            self.minio1.get_object(Bucket="test_upload", Key="2.txt")
+
+    def test_delete_objects(self):
+        self.minio2.create_bucket(Bucket="test-delete-objects")
+        content = "TEST_DELETE_OBJECTS"
+        self.botwo_client.put_object(Bucket="test-delete-objects", Key="test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-delete-objects", Key="test/2.txt", Body=content)
+
+        res = self.botwo_client.delete_objects(Bucket="delete-objects1", Delete={
+            'Objects': [
+                {
+                    'Key': '1.txt',
+                },
+                {
+                    'Key': '2.txt',
+                },
+            ]
+        })
+        self.assertEqual(res["Deleted"][0]["Key"], "test/1.txt")
+        self.assertEqual(res["Deleted"][1]["Key"], "test/2.txt")
+
+    def test_paginator(self):
+        self.minio2.create_bucket(Bucket="test-paginator")
+        content = "TEST_PAGINATOR"
+        self.botwo_client.put_object(Bucket="test-paginator", Key="dev/test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-paginator", Key="dev/test/2.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-paginator", Key="1.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-paginator", Key="test/3.txt", Body=content)
+
+        self.assertEqual(self.botwo_client.can_paginate(operation_name='list_objects'), True)
+
+        # Create a reusable Paginator
+        paginator = self.botwo_client.get_paginator(operation_name='list_objects')
+
+        # Create a PageIterator from the Paginator
+        page_iterator = paginator.paginate(Bucket='pagi', Prefix="test")
+
+        # Compare the iterators
+        for page in page_iterator:
+            self.assertEqual(len(page['Contents']), 2)
+
+    def test_list_objects(self):
+        self.minio2.create_bucket(Bucket="test-list-objects")
+        content = "TEST_LIST_OBJECTS"
+
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="dev/test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="dev/test/2.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="dev/1.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="1.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="test/3.txt", Body=content)
+
+        objects = self.botwo_client.list_objects(Bucket="list", Prefix="test")
+
+        self.assertEqual(len(objects['Contents']), 2)
+        # test empty prefix
+        objects = self.botwo_client.list_objects(Bucket="list", Prefix="")
+        self.assertEqual(len(objects['Contents']), 3)
```

Comment:
I think we need to test the return values here -- they may need to be edited back to the correct prefix; see https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_Example_3 for what responses look like.

This can be left as an open issue to do in a separate PR; just open the issue and mark # BUG(#123): ... to link the bug number.

Reply:
Done

Reply:
This remains open.